### ISSUE TITLE
collect-debug-info gets 405 on gist upload

### ISSUE THREAD

**Issue by temujin9** [flynn/flynn#2555]
```
INFO[03-10|22:08:15] uploading logs and debug information to a private, anonymous gist 
INFO[03-10|22:08:15] this may take a while depending on the size of your logs 
INFO[03-10|22:08:15] getting flynn-host logs 
INFO[03-10|22:08:24] getting job logs 
INFO[03-10|22:10:27] getting system information 
INFO[03-10|22:10:36] creating anonymous gist 
238.87 MB / 238.87 MB [===========================================================================================================================================================================================================================] 100.00 % 48.63 MB/s 0EROR[03-10|22:10:51] unexpected HTTP status: 405 
238.87 MB / 238.87 MB [=========================================================================================================================================================================================================================] 100.00 % 15.86 MB/s 15s
22:10:51.700288 host.go:141: unexpected HTTP status: 405
```

This is a semi-intermittent issue, even on the same host.

**Comment by titanous** [comment]
This probably based on the file size, we should bail out before attempting the upload if it is greater than the limit.

**Comment by temujin9** [comment]
This is happening quite regularly now. Rather than randomly aborting, it would be better to a) found a new place to upload to, or b) trimmed down to fit the Gist maximum.

**Comment by titanous** [comment]
Yeah. I like your proposal to have an option to upload to S3.

**Comment by philiplb** [comment]
S3 or something similar would be good as the Github support recently complained due to the high amount of deletion requests I sent them. :-)

**Comment by titanous** [comment]
This was fixed in #2912. A new issue should be opened if S3 support is still wanted.
